src/app/core/store/collections/collections.action.ts

```diff
@@ -0,0 +1,240 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Action} from '@ngrx/store';
+import {QueryModel} from '../navigation/query.model';
+import {AttributeModel, CollectionModel} from './collection.model';
+import {PermissionModel, PermissionType} from '../permissions/permissions.model';
+
+export enum CollectionsActionType {
+
+  GET = '[Collections] Get',
```

Comment:
Are we using the fact that it's a `string`? If not, this could be made `Symbol` for extra bit of safety and performance.
http://www.zsoltnagy.eu/es6-symbols-and-its-use-cases/

Reply:
Yes, NgRx action types actually need to be strings (see [Action](https://github.com/ngrx/platform/blob/master/modules/store/src/models.ts#L1-L3) interface which is implemented by all our actions).

In the past, it was recommended to use string constants for actions types. However, we use new version of TypeScript which supports string enums so why not make use of them?